### PR TITLE
Modernize zephyr hello world sample

### PR DESCRIPTION
**Pull request by SebastianBoe**
This PR includes several patches to modernize the Zephyr hello-world application according to the latest best-practices.

### CODE REVIEW

File: samples/zephyr/hello-world/CMakeLists.txt [hunk 1]
```diff
@@ -14,17 +14,6 @@ if (NOT DEFINED BOARD)
   set(BOARD frdm_k64f)
 endif()
 
-# The default top-level application configuration is prj.conf.
-# You can place additional board-specific files in boards/${BOARD}.conf,
-# and they will be merged into the configuration along with prj.conf.
-macro(set_conf_file)
-  if(EXISTS ${APPLICATION_SOURCE_DIR}/boards/${BOARD}.conf)
-    set(CONF_FILE "prj.conf ${APPLICATION_SOURCE_DIR}/boards/${BOARD}.conf")
```
Comment (mbolivar) [inline samples/zephyr/hello-world/CMakeLists.txt]: The per-board .conf file is in addition to prj.conf; it doesn't replace it. The Zephyr prj_BOARD.conf convention is for standalone conf files tat target a single board. The boards/BOARD.conf files are mix-ins. I think this patch should be dropped.

**Comment by d3zd3z** [comment]
Agreed with Marti about the per-board mixin config. Otherwise, looks good.

Also, I believe the preferred English spelling is "existent".

File: samples/zephyr/hello-world/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply samples/zephyr/hello-world/CMakeLists.txt]: Aha, my mistake, will fix.

Comment (SebastianBoe) [reply samples/zephyr/hello-world/CMakeLists.txt]: Fixed.

**Comment by SebastianBoe** [comment]
All review feedback is now resolved.

**Comment by carlescufi** [comment]
Looks good to me

**Pull request merged**